python/kserve/kserve/storage.py

```diff
@@ -77,6 +78,8 @@ def download(uri: str, out_dir: str = None) -> str:
             # Don't need to download models if this InferenceService is running in the multi-model
             # serving mode. The model agent will download models.
             return out_dir
+        elif uri.startswith(_STROAGE_SPEC_PREFIX):
```

Comment:
instead of using `secret://`, can we check if the secret volume exists and then load the volume, download using storage spec?

Reply:
Sure, I can do that. My original intent is try to not break any backward compatibility until we introduced all storage support in the new storage spec.

Reply:
removed